x-pack/test/functional/es_archives/endpoint/metadata/destination_index/data.json

```diff
@@ -142,7 +142,17 @@
       },
       "agent": {
         "id": "963b081e-60d1-482c-befd-a5815fa8290f"
-      }
+      },
+      "Endpoint": {
```

Comment:
@pzl @kevinlog here's my theory for what happened:

When I updated `dockerImage`, it pulled in the package from https://github.com/elastic/endpoint-package/pull/124.
In that package `HostDetails.Endpoint` was removed from the index template and wasn't searchable anymore. In this PR, I updated the test data so it was consistent with the package.

I also noticed that `api_feature/data.json` looks like it was already updated, but `destination_index/data.json` doesn't look like it was. I didn't remove anything, from the, data just copied `HostDetails.Endpoint.*` at `Endpoint.*`